src/cfnlint/core.py

```diff
@@ -185,17 +201,30 @@ def run_checks(filename, template, rules, regions, mandatory_rules=None):
                 unsupported_regions, REGIONS)
             raise InvalidRegionException(msg, 32)
 
-    matches = []
+    errors = []
+
+    runner = cfnlint.runner.Runner(rules, filename, template, regions,
+                                   mandatory_rules=mandatory_rules)
+
+    # Transform logic helps with handling serverless templates
+    ignore_transform_error = False
+    if not rules.is_rule_enabled(TransformError()):
+        ignore_transform_error = True
+
+    errors.extend(runner.transform())
+
+    if errors:
+        if ignore_transform_error:
```

Comment:
```suggestion
    errors.extend(runner.transform())

    if errors:
        if not rules.is_rule_enabled(TransformError()):
```

Reply:
yes we could.  Could put a deprecation notice on it.